src/components/NavItem/SubLink/index.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+
+import { createPolymorphicComponent } from '@mantine/core'
+import classNames from 'classnames'
+
+import styles from './SubLink.module.scss'
+
+export interface SubLinkProps {
+  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+  component?: any
+  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+  children?: any
```

Comment:
these can be ReactNode

Reply:
updated children, but for component, Mantine uses `any` https://github.com/mantinedev/mantine/blob/fbcee929e0b11782092f48c1e7af2a1d1c878823/packages/%40mantine/core/src/core/Box/Box.tsx#L65. I think it is probably fine to just match them since we already know that works with next/link, react router, etc.